### PR TITLE
Log joining rejection reason.

### PR DESCRIPTION
**Pull request by kriscieslak**

### CODE REVIEW

File: modules/sdk/src/main/scala/org/tessellation/sdk/http/p2p/clients/SignClient.scala [hunk 1]
```diff
@@ -24,12 +29,17 @@ object SignClient {
   def make[F[_]: Async: SecurityProvider](client: Client[F]): SignClient[F] =
     new SignClient[F] with Http4sClientDsl[F] {
 
+      private val logger = Slf4jLogger.getLogger
```
Comment (marcinwadon) [inline modules/sdk/src/main/scala/org/tessellation/sdk/http/p2p/clients/SignClient.scala:32]: `getLogger[F]`

File: modules/sdk/src/main/scala/org/tessellation/sdk/http/p2p/clients/SignClient.scala [hunk 2]
```diff
@@ -24,12 +29,17 @@ object SignClient {
...
+
       def getRegistrationRequest: PeerResponse[F, RegistrationRequest] =
         PeerResponse[F, RegistrationRequest]("registration/request")(client)
 
       def joinRequest(jr: JoinRequest): PeerResponse[F, Boolean] =
         PeerResponse[F, Boolean]("cluster/join", POST)(client) { (req, c) =>
-          c.successful(req.withEntity(jr))
+          c.run(req.withEntity(jr)).use {
+            case Status.Successful(_) => Applicative[F].pure(true)
+            case res                  => res.as[String].flatTap(msg => logger.warn(s"Join request rejected due to: $msg")).map(_ => false)
```
Comment (marcinwadon) [inline modules/sdk/src/main/scala/org/tessellation/sdk/http/p2p/clients/SignClient.scala:41]: `.map(_ => false)` = `.as(false)`

**Comment by marcinwadon** [comment]
@kriscieslak is it meant to be merged to the other branch than `develop`?

**Comment by kriscieslak** [comment]
Yes, I wanted it to be merged to `joining-fix-discover` to preserve order of changes in which they are merged with develop, after `joining-fix-discover` is merged to developed github will automatically update this PR to be merged to `develop`. 
Though after closer look, these changes are independent so no need for such approach. I could have create PR directly to `develop`.

**Review by marcinwadon — DISMISSED** [review]

**Review by kpudlik — APPROVED** [review]

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**